compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/parser/BLangAnonymousModelHelper.java

```diff
@@ -97,6 +101,34 @@ public String getNextAnonymousTypeKey(PackageID packageID) {
         return ANON_TYPE + UNDERSCORE + nextValue;
     }
 
+    public String getNextAnonymousTypeKey(PackageID packageID, Stack<String> suffixes) {
+        if (suffixes.isEmpty()) {
+            return getNextAnonymousTypeKey(packageID);
+        }
+        return createAnonTypeName(suffixes, packageID);
+    }
+
+    private String createAnonTypeName(Stack<String> suffixes, PackageID pkgId) {
+        String name = ANON_TYPE;
+        for (int i = suffixes.size() - 1; i >= 0; i--) {
+            name = name.concat(suffixes.elementAt(i) + DOLLAR);
+        }
+        if (!anonTypesNamesPerPkg.containsKey(pkgId)) {
+            Map<String, Integer> anonTypesName = new HashMap<>();
+            anonTypesName.put(name, 1);
+            anonTypesNamesPerPkg.put(pkgId, anonTypesName);
+            return name + UNDERSCORE + 0;
+        }
+        Map<String, Integer> anonTypesNames = anonTypesNamesPerPkg.get(pkgId);
+        if (!anonTypesNames.containsKey(name)) {
+            anonTypesNames.put(name, 1);
+            return name + UNDERSCORE + 0;
```

Comment:
Is this okay that `id` is 1 but the name has `0` at the end?

Reply:
Changed